presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java

```diff
@@ -166,16 +177,15 @@ public Page getOutput()
                     break;
                 }
             }
-        }
 
-        // only flush full pages unless we are done
-        if (pageBuilder.isFull() || (finishing && !pageBuilder.isEmpty() && probe == null)) {
-            Page page = pageBuilder.build();
-            pageBuilder.reset();
-            return page;
+            // only flush full pages unless we are done
+            if (readyToBuildOutputPage()) {
+                outputPage = probe.buildDictionaryPage(probeIndexBuilder.getArray(), buildPageBuilder);
+                reset();
+            }
         }
 
-        return null;
+        return outputPage;
```

Comment:
I preferred `return null;` better. It clearly shows where the output can be produced. Please inline `outputPage`